stdlib/database/sql/src/main/java/org/ballerinalang/database/sql/SQLDataIterator.java

```diff
@@ -526,11 +538,13 @@ private void handleStringValue(String stringValue, String fieldName,
         int fieldTypeTag = fieldType.getTag();
         if (fieldTypeTag == TypeTags.UNION_TAG) {
             BRefType refValue = stringValue == null ? null : new BString(stringValue);
-            validateAndSetRefRecordField(bStruct, fieldName, TypeTags.STRING_TAG,
+            int[] expectedTypeTags = { TypeTags.STRING_TAG, TypeTags.JSON_TAG };
```

Comment:
Any and Anydata should be also part of this?

Reply:
As discussed offline we'll do the complete fix (which requires some refactoring) in a separate PR and merge this considering the urgency of the fix.